### PR TITLE
Remove features that has been marked as deprecated

### PR DESCRIPTION
**Pull request by jepebe**
Many features has for a longer (or shorter) time been marked as deprecated. This PR resolves some of them.

This resolves some of the issues mentioned in #961

### CODE REVIEW

**Comment by ertomatic** [comment]
Can one of the admins verify this patch?

**Comment by jepebe** [comment]
Jenkins build this please

**Comment by jepebe** [comment]
Jenkins retest this please

File: python/res/enkf/enkf_main.py [hunk 1]
```diff
@@ -229,47 +222,17 @@ def _init_res_config(self, config):
...
-            res_config.convertToCReference(self)
-
-            return res_config
+        if config is None:
+            config = ResConfig(None)
+            config.convertToCReference(self)
+            return config
 
-        raise TypeError("Expected ResConfig, received: %r" % res_config)
+        raise TypeError("Expected ResConfig, received: %r" % config)
```
Comment (sondreso) [inline python/res/enkf/enkf_main.py:230]: Nitpick, but maybe switch to the new formatting: `raise TypeError("Expected ResConfig, received: {}".format(config))`

**Review by sondreso — APPROVED** [review]
Except for the very nitpicky comment this looks very good :rocket:

**Pull request merged**